### DISCUSSION TITLE
How to check for cookie

### DISCUSSION THREAD

**Discussion by Onnyxx1** [statamic/cms#6728] [General]
I want to check if a cookie is set in my antlers template. My project is based on laravel and uses statamic as frontend.
What is the correct way to check for a cookie?

**Comment by duncanmcclean** [comment]
You can make [your own tag](https://statamic.dev/extending/tags) to get a user's cookies, then use that to check if the cookie you're looking for exists. (getting the cookies itself is a Laravel thing)

You could also use JavaScript to get a user's cookies, it's possible that may work too, depending on your use case.

**Comment by Onnyxx1** [comment]
Thanks for the hint with the tag, I can read the cookie with that.
But now, I have another problem: It seems, that the value of the cookie is only read once after I save and publish the page in the control panel. When I change the value of the cookie via browser inspector, the old value is still displayed after I refresh the page. Is there a way to change that behaviour?

**Reply by torbentschechne** [reply]
Hey there, how did you solve it? 

I tried using it like this in my `<head>`

```
{{ if { cookie:has key="laravel_cookie_consent"} }}
    <script
        id="space-widget-script"
        data-auto-init
        src="https://cdn.....js"
        async
    />
  {{ /if }}
```

But somehow this is not working.
